src/main/c/ssl.c

```diff
@@ -1201,6 +1202,11 @@ TCN_IMPLEMENT_CALL(jlong /* SSL * */, SSL, newSSL)(TCN_STDARGS,
         tcn_ThrowException(e, "cannot create new ssl");
         return 0;
     }
+
+    // Store the handshakeCount in the SSL instance.
+    *handshakeCount = 0;
```

Comment:
Yes I think that is a good idea in general...like I said if we can't allocate a `uint16_t` we are likely near death anyways :)